apps/src/code-studio/components/Libraries/libraryParser.js

```diff
@@ -0,0 +1,80 @@
+import {getFunctionsAndMetadata} from '@cdo/apps/lib/tools/jsinterpreter/JSInterpreter';
+
+/**
+ * Finds all functions in the provided code and returns them with any metadata
+ * @param {string} code The code to parse to get all functions
+ * @returns {array} A list of objects where each object represents a function with its doc comment and params
+ */
+export function getFunctions(code) {
+  return getFunctionsAndMetadata(code);
+}
+
+/**
+ * Takes the input library elements and creates a json string from them to be
+ * uploaded to S3
+ * @param {string} code The code that makes up the library
+ * @param {array} selectedFunctions The list of functions that will be exported from the library
+ * @param {string} libraryName The name of the library to be exported
+ * @returns {string} A JSON string representign the library
+ */
+export function createLibraryJson(code, selectedFunctions, libraryName) {
+  if (
```

Comment:
I wonder if it would make sense to put this through something like the apiValidateType()? https://github.com/code-dot-org/code-dot-org/blob/de0abcd90d7bcfe1929e60e44a237ff25427832a/apps/src/lib/util/javascriptMode.js#L38

Reply:
I think I'm going to leave as is since this is more targeted toward student code. Or were you suggesting I create a new util library for input validation?

Reply:
Thinking deeper on it, the benefit of apiValidateType is that it's a central place we can use to validate the student code that comes in for different commands. For this case, it's really this one place where we want to validate the student code. I don't think it's worth blocking this PR on it.